packages/kamino-sdk/src/Kamino.ts

```diff
@@ -6602,17 +6600,24 @@ export class Kamino {
       let amountADecimal = new Decimal(0);
       let amountBDecimal = new Decimal(0);
       if (bin) {
-        amountADecimal = new Decimal(bin.amountX.toString());
-        amountBDecimal = new Decimal(bin.amountY.toString());
+        if (!bin.amountX.eq(new BN(0))) {
+          amountADecimal = new Decimal(bin.amountX.toString());
+        }
+        if (!bin.amountY.eq(new BN(0))) {
+          amountBDecimal = new Decimal(bin.amountY.toString());
+        }
       }
+
       const ratio = amountADecimal.div(amountBDecimal);
```

Comment:
This will not work correctly if amountBDecimal is 0! which is entirely possible if bin is not defined above.. which means this will be defined as Infinity